### DISCUSSION TITLE
Avoid parallelization of different assemblies

### DISCUSSION THREAD

**Discussion by danrot** [xunit/xunit#2510] [Question] [answered]
I am building a small CMS for educational purposes in my [GitHub Repositoy](https://github.com/danrot/DotNetCMS). I split that into multiple different C# projects with separate test projects.

When I am running the `dotnet test` command in each test project separately everything works as expected. If I run the `dotnet test` in the root folder of the GitHub repository that only works once, afterwards I am getting some database related errors. This errors are not always the same and I strongly suspect that this is a parallelization issue. Sometimes I get a MySQL error regarding a deadlock, the other time it says that it tries to delete a database that does not exist, ...

I guess the reason for that is that there are two tests (https://github.com/danrot/DotNetCMS/blob/master/DotNetCMS.Persistence.Tests/Pages/EntityFrameworkCorePageRepositoryTest.cs#L20 and https://github.com/danrot/DotNetCMS/blob/master/DotNetCMS.Program.Tests/PagesControllerTest.cs#L31) in different assemblies trying to delete the database, and that these two tests are ran concurrently. This would at least explain why they are working for the first time (where the build process might delay the tests) and when I run them in isolation.

If I am understanding the [configuration documentation about `parallelizeAssembly`](https://xunit.net/docs/configuration-files#parallelizeAssembly), then these assemblies should not be running in parallel, and therefore there should not be any issues. However, even if I [add the `xunit.runner.json` file as described including the change in the `.csproj` file](https://xunit.net/docs/configuration-files#file) for both C# projects and set `parallelizeTestCollections`and `parallelizeAssembly` to `false` it is still not working. It seems to me that the `xunit.runner.json` file is not recognized at all, since even if I add settigs like `diagnosticMessages` nothing seems to change.

Feel free to checkout my code at my repository to help me with this issue: https://github.com/danrot/DotNetCMS

All I want to manage, is to be able to run `dotnet test` multiple times without failing it at any instance.

**Comment by bradwilson** [comment, marked as answer]
This is VSTest doing this, not us.

See a similar post on the NUnit project about how to disable parallelization in VSTest with a RunSettings file: https://github.com/nunit/nunit3-vs-adapter/issues/657#issuecomment-531374774

**Reply by bradwilson** [reply]
Actually, based on reading more of the comments, it seems like the best answer is this:

```
dotnet test -m:1 YourSolutionFile.sln
```
